tests/ccgbugs/t15623.nim

```diff
@@ -0,0 +1,11 @@
+discard """
+  action: "compile"
+"""
+
+# bug #15623
+block:
+  echo cast[ptr int](nil)[]
```

Comment:
you can remove
```
discard """
  action: "compile"
"""
```
(hence make the test megatest-joinable, and simpler to use also) as follows:

```nim
block:
  if false:
   discard cast[ptr int](nil)[]
```

Reply:
Done. Thanks!